documentation/proposals/Proposal - 3.0 & 3.X Software Development Plan.md

```diff
@@ -0,0 +1,149 @@
+# Summary
+
+3.0 software development plan & ongoing monthly update, breaking change, and support policy for 3.X.
+
+# Contributors
+- Dylan P (@Perksey)
+
+# Current Status
+- [x] Proposed
+- [ ] Discussed with Working Group
+- [ ] Approved
+- [ ] Implemented
+
+# Silk.NET 3.0
+
+## Goals of 3.0
+
+The key tenets of 3.0 are **portability**, **maintainability**, **usability**, and **performance**. To this end, the following objectives have been identified:
+- Use .NET 6 - the first version of modern .NET to run on the majority of our desired target platforms
+    - (tenet: portability) 
+- Allow Silk.NET's rich abstractions to be integrated into other frameworks rather than being completely standalone.
+    - In order of priority: WPF, WinForms, MAUI, Avalonia (based on demand). (tenet: usability) 
+- Rewrite windowing to be more portable and facilitate true write-once-run-everywhere.
+    - For more information, see [the Windowing 3.0 proposal](Proposal - Windowing 3.0.md). (tenet: portability)
+- Remove the bulk of our bindings generation code in favour of more mature alternatives
+    - For more information, see [the SilkTouch 3.0 proposal](Proposal - Generation of Library Sources and PInvoke Mechanisms.md). (tenet: maintainability)
+- Accelerate our maths library using SIMD hardware intrinsics
+    - For more information, see [the Vectorization SIMD proposal](Proposal - Vectorization - SIMD.md). (tenet: performance)
+- Enhance our input library to add more functionality and future-proof it against future functionality additions.
+    - For more information, see [the Enhanced Input Events proposal](Proposal - Enhanced Input Events.md). (tenet: maintainability)
+- Refactor our input library to work in multiple scenarios and environments
+    - For more information, see [the Multi-Backend Input proposal](Proposal - Multi-Backend Input.md). (tenet: usability)
+
+Silk.NET 3.0 presents us with an opportunity to rethink the entire library taking into account everything we've learnt over the past 2 years of the project's development.
+
+## Development Roadmap
+
+Note that this development roadmap does not take into account unit tests, only functional tests such as experiments. The team should of course stride to add as many tests as possible where possible.
```

Comment:
Why. One of the important things for 3.0 imo is that it cements it as a production ready product. Unit tests are vital.

Reply:
I don't expect us to do fully functional testing, but I've found a very nice looking lib for approval tests. We can very likely get some actual unit testing working with not as much effort. Imo if we can't test it it's not good. Virtually all of our library is at the heart of a user's library, if we have a bug we have a bug at the heart of the users app.

Reply:
Oh yeah for sure, I do really want unit testing but I'm also being realistic about our time constraints - we are starting 3.0 development incredibly late!